src/connector/src/source/kinesis/source/message.rs

```diff
@@ -14,16 +14,25 @@
 
 use aws_sdk_kinesis::types::Record;
 use aws_smithy_types_convert::date_time::DateTimeExt;
+use risingwave_common::types::{DatumRef, ScalarRefImpl};
 
 use crate::source::{SourceMessage, SourceMeta, SplitId};
 
 #[derive(Clone, Debug)]
 pub struct KinesisMeta {
-    // from `approximate_arrival_timestamp` of type `Option<aws_smithy_types::DateTime>`
-    #[expect(dead_code)]
     timestamp: Option<i64>,
```

Comment:
Why do we store the intermediate type here `aws_smithy_types::DateTime -> i64_micros -> Timestamptz`? Is it better to encapsulate the conversion to a single place?

| KinesisMeta | from_kinesis_record | extract_timestamp |
|-|-|-|
|i64|aws_smithy_types::DateTime -> i64|i64 -> Timestamptz|
|aws_smithy_types::DateTime|noop|aws_smithy_types::DateTime -> Timestamptz|
|Timestamptz|aws_smithy_types::DateTime -> Timestamptz|noop|